### PR TITLE
chore: Refactor ModalRules to use Dialog component

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

Converts ModalRules to use the new Dialog component with Event-based trigger.
This is the final legacy Modal to be converted, so this PR also includes removal of the old Modal component and its related files.

### CODE REVIEW

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx [hunk 1]
```diff
@@ -39,44 +38,53 @@ export const ModalFormRules = ({
...
-  const [choiceRules, setChoiceRules] = useState(initialChoiceRules);
+  const [choiceRules, setChoiceRules] = useState<ChoiceRule[]>(initialChoiceRules);
+
+  useEffect(() => {
+    choiceRulesRef.current = choiceRules;
+  }, [choiceRules, choiceRulesRef]);
+
+  let sortedElements = sortByLayout({ layout: form.layout, elements: elements });
+
+  if (groupsEnabled) {
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx:62]: Thinking we don't need this check now thoughts ?

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
+import { ChoiceRule, getElementsWithRuleForChoice } from "@lib/formContext";
 import { ConditionalSelector } from "@formBuilder/components/shared/conditionals/ConditionalSelector";
 import { sortByGroups, sortByLayout } from "@lib/utils/form-builder";
 import { AddOther } from "@formBuilder/components/shared/conditionals/AddOther";
 
 import Markdown from "markdown-to-jsx";
+import { FormElementWithIndex } from "@lib/types/form-builder-types";
 
-export const ModalFormRules = ({
+export const Rules = ({
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx:15]: Maybe we should call this RulesSelectors or something ... to better describe what the UI is

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  const { t, i18n } = useTranslation("form-builder");
+  const { Event } = useCustomEvent();
+  const dialog = useDialogRef();
+  const { togglePanel } = useTreeRef();
+  const router = useRouter();
+  const { refs } = useRefsContext();
+
+  const [mode, setMode] = React.useState<"add" | "edit">("add");
+  const [item, setItem] = React.useState<FormElementWithIndex | undefined>(undefined);
+  const [optionId, setOptionId] = React.useState<string | null>(null);
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx:37]: Should we call this selectedOptionId ?

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      setIsOpen(false);
+      return;
+    }
+
+    const els = choiceRulesToConditonalRules(elements, choiceRulesRef.current);
+    // Update the element rules
+    Object.entries(els).forEach(([key, value]) => {
+      const element = elements.find((el) => el.id == Number(key));
+      if (!element) return;
+      const properties = { ...element.properties, conditionalRules: value };
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx:92]: value should probably be  rules here

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx (hunk 2 above)
Comment (dsamojlenko) [reply app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx]: Renamed to RulesForm 33b04179a37334558bd7784a0266c712c3edc635

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+        // Update the element rules if they have changed
+        const properties = { ...el.properties, conditionalRules: rules };
+        updateField(getPathString(el.id, elements), properties);
+      });
+      setIsOpen(false);
+      return;
+    }
+
+    const els = choiceRulesToConditonalRules(elements, choiceRulesRef.current);
```
[the remark this answers is not in the document]
Comment (dsamojlenko) [reply app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx]: f49a14ad72f55c95f16f4ef5410858b8bd9e8570

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx (hunk 4 above)
Comment (dsamojlenko) [reply app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx]: f142e551781be5a10e73d6fbf43586a600360f73

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx (hunk 1 above)
Comment (dsamojlenko) [reply app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx]: nope d95ae58402224e375d78a2754d470cf4fe856d4e

**Review by timarney — APPROVED** [review]

**Pull request merged**